<!DOCTYPE html>
<html lang="en">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta charset="utf-8">
    <title>VcfFilter</title>

    <!-- Latest compiled and minified CSS -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <!-- Optional theme -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css" integrity="sha384-rHyoN1iRsVXV4nD0JutlnGaslCJuC7uwjduW9SVrLvRYooPp2bWYgmgJQIXwl/Sp" crossorigin="anonymous">

    <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>

    <!-- Latest compiled and minified JavaScript -->
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

    <!--[if lt IE 9]>
      <script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link href="css/docs.css" rel="stylesheet">

  </head>

  <body>
  <nav class="navbar navbar-default navbar-fixed-top">
    <div class="container-fluid">
      <div class="navbar-header">
        <div class="navbar-header">
          <a class="navbar-brand" href="#">VcfFilter</a>
          <ul class="nav navbar-nav navbar-right">
            <li class="navbar-text" id="Version">
              Version</li>
          </ul>
        </div>
      </div>



    </div>

  </nav>
  <div class="container">

    <div class="row">
      <div class="col-sm-3 col-md-2 sidebar" >

        <ul class="nav nav-sidebar nav-list">
          <li><b>Index</b></li>
          
          <li><a href="#description"><span>Description</span></a></li>
          
          <li><a href="#installation"><span>Installation</span></a></li>
          
          <li><a href="#manual"><span>Manual</span></a></li>
          
          <li><a href="#about"><span>About</span></a></li>
          
          <li><a href="#contributing"><span>Contributing</span></a></li>
          
          <li><a href="#contact"><span>Contact</span></a></li>
          
        </ul>
        <ul class="nav nav-sidebar nav-list">
          <li><b>Project Links</b></li>

          <li><a href="https://github.com/biopet/vcffilter">Source code</a></li>
          <li><a href="https://github.com/biopet/vcffilter/issues">Feedback, feature requests and bug reports</a></li>
          <li><a href="api/index.html">API Documentation</a></li>
        </ul>
        <ul class="nav nav-sidebar nav-list">
          <li><b>Documentation</b></li>
          <li><a href="../index.html">Latest stable version</a></li>
          <li><a href="../develop/index.html">Development version</a></li>
        </ul>
      </div>
      
      <div class="col-sm-9 col-sm-offset-3 col-md-10 col-md-offset-2 main" id="top">

        
        <h1 id="description" class="section">Description</h1>
        <p>This tool enables a user to filter VCF files. For example on sample depth and/or total depth. It can also be used to
        filter out the reference calls and/or minimum number of sample passes. There is a wide set of options which one can
        use to change the filter settings.</p>
        
        <h1 id="installation" class="section">Installation</h1>
        <p>VcfFilter requires Java 8 to be installed on your device. Download Java 8
        <a href="http://www.oracle.com/technetwork/java/javase/downloads/jre8-downloads-2133155.html">here</a>
        or install via your distribution&#39;s package manager.</p>
        <p>Download the latest version of VcfFilter <a href="https://github.com/biopet/vcffilter/releases/">here</a>.
        To generate the usage run:</p>
        <pre><code>java -jar &lt;VcfFilter_jar&gt; --help</code></pre>
        
        <h1 id="manual" class="section">Manual</h1>
        <p>This tool filters VCF files on a number of values. For example, it can filter on sample depth and/or total depth. It
        can also filter out the reference calls and/or minimum number of sample passes. For more on filtering options and how
        to set them, please refer to the help usage.</p>
        
        <h2 id="example" class="section">Example</h2>
        <p>To filter a VCF for variants with a minimum quality score of 50:</p>
        <pre><code>java -jar &lt;VcfFilter_jar&gt; \
-I input.vcf \
-o output.vcf \
--minQualScore 50</code></pre>
        
        <h2 id="usage" class="section">Usage</h2>
        <p>Usage for VcfFilter:</p>
        <p>&lt;table class=&quot;table&quot;&gt;
          &lt;thead&gt;
            &lt;tr&gt;
              &lt;th&gt;Option&lt;/th&gt;
              &lt;th&gt;Required&lt;/th&gt;
              &lt;th&gt;Can occur multiple times&lt;/th&gt;
              &lt;th&gt;Description&lt;/th&gt;
            &lt;/tr&gt;
          &lt;/thead&gt;
          &lt;tbody&gt;
            &lt;tr&gt;
              &lt;td&gt;--log<em>level, -l&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Level of log information printed. Possible levels: &#39;debug&#39;, &#39;info&#39;, &#39;warn&#39;, &#39;error&#39;&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--help, -h&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Print usage&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--version, -v&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Print version&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--inputVcf, -I&lt;/td&gt;
              &lt;td&gt;yes&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Input vcf file&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--outputVcf, -o&lt;/td&gt;
              &lt;td&gt;yes&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Output vcf file&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--invertedOutputVcf&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;inverted output vcf file&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--minSampleDepth&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Min value for DP in genotype fields&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--minTotalDepth&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Min value of DP field in INFO fields&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--minAlternateDepth&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Min value of AD field in genotype fields&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--minSamplesPass&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Min number off samples to pass --minAlternateDepth, --minBamAlternateDepth and --minSampleDepth&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--resToDom&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Only shows variants where child is homozygous and both parants hetrozygous&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--trioCompound&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Only shows variants where child is a compound variant combined from both parants&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--deNovoInSample&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Only show variants that contain unique alleles in complete set for given sample&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--deNovoTrio&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Only show variants that are denovo in the trio&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--trioLossOfHet&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Only show variants where a loss of hetrozygosity is detected&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--mustHaveVariant&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Given sample must have 1 alternative allele&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--mustNotHaveVariant&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Given sample may not have alternative alleles&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--calledIn&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Must be called in this sample&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--mustHaveGenotype&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Must have genotoype &lt;genotype&gt; for this sample. Genotype can be NO</em>CALL, HOM<em>REF, HET, HOM</em>VAR, UNAVAILABLE, MIXED&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--diffGenotype&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Given samples must have a different genotype&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--filterHetVarToHomVar&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;If variants in sample 1 are heterogeneous and alternative alleles are homogeneous in sample 2 variants are filtered&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--filterRefCalls&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Filter when there are only ref calls&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--filterNoCalls&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Filter when there are only no calls&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--uniqueOnly&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Filter when there more then 1 sample have this variant&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--sharedOnly&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Filter when not all samples have this variant&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--minQualScore&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;Min qual score&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--id&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;Id that may pass the filter&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--idFile&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;File that contain list of IDs to get from vcf file&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--minGenomeQuality&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;The minimum value in the Genome Quality field.&lt;/td&gt;
            &lt;/tr&gt;
            &lt;tr&gt;
              &lt;td&gt;--advancedGroups&lt;/td&gt;
              &lt;td&gt;no&lt;/td&gt;
              &lt;td&gt;yes (unlimited)&lt;/td&gt;
              &lt;td&gt;All members of groups sprated with a &#39;,&#39;&lt;/td&gt;
            &lt;/tr&gt;
          &lt;/tbody&gt;
        &lt;/table&gt;</p>
        
        <h1 id="about" class="section">About</h1>
        <p>VcfFilter is part of BIOPET tool suite that is developed at LUMC by <a href="http://sasc.lumc.nl/">the SASC team</a>.
        Each tool in the BIOPET tool suite is meant to offer a standalone function that can be used to perform a
        dedicate data analysis task or added as part of <a href="http://biopet-docs.readthedocs.io/en/latest/">BIOPET pipelines</a>.</p>
        <p>All tools in the BIOPET tool suite are <a href="https://www.gnu.org/philosophy/free-sw.html">Free/Libre</a> and
        <a href="https://opensource.org/osd">Open Source</a> Software.</p>
        
        <h1 id="contributing" class="section">Contributing</h1>
        <p>The source code of VcfFilter can be found <a href="https://github.com/biopet/vcffilter">here</a>.
        We welcome any contributions. Bug reports, feature requests and feedback can be submitted at our
        <a href="https://github.com/biopet/vcffilter/issues">issue tracker</a>.</p>
        <p>VcfFilter is build using sbt. Before submitting a pull request, make sure all tests can be passed by
        running <code>sbt test</code> from the project&#39;s root. We recommend using an IDE to work on VcfFilter. We have had
        good results with <a href="https://www.jetbrains.com/idea/">this IDE</a>.</p>
        
        <h1 id="contact" class="section">Contact</h1>
        <p>&lt;p&gt;
          &lt;!-- Obscure e-mail address for spammers --&gt;
        For any question related to VcfFilter, please use the
        &lt;a href=&#39;https://github.com/biopet/vcffilter/issues&#39;&gt;github issue tracker&lt;/a&gt;
        or contact
         &lt;a href=&#39;http://sasc.lumc.nl/&#39;&gt;the SASC team&lt;/a&gt; directly at: &lt;a href=&#39;&amp;#109;&amp;#97;&amp;#105;&amp;#108;&amp;#116;&amp;#111;&amp;#58;&amp;#115;&amp;#97;&amp;#115;&amp;#99;&amp;#64;&amp;#108;&amp;#117;&amp;#109;&amp;#99;&amp;#46;&amp;#110;&amp;#108;&#39;&gt;
        &amp;#115;&amp;#97;&amp;#115;&amp;#99;&amp;#64;&amp;#108;&amp;#117;&amp;#109;&amp;#99;&amp;#46;&amp;#110;&amp;#108;&lt;/a&gt;.
        &lt;/p&gt;</p>      

      </div>
    </div>

  </div>
  <script>
    var loc = window.location.pathname;
    var thisDir = loc.substring(0, loc.lastIndexOf('/'));
    var splitLocation = window.location.pathname.split('/');
    var version = splitLocation[splitLocation.length - 2]
    document.getElementById("Version").innerHTML = "Version " + version;
  </script>
</body>

</html>
